### DISCUSSION TITLE
ESP01-S Relay borad

### DISCUSSION THREAD

**Discussion by vifi42** [RavenSystem/esp-homekit-devices#2085] [MEPLHAA Scripts]
Hey! I'm trying to use a very simple Relay-Board with an ESP01-S-Controller. It has a relay connected to GPIO0 and I'm simply trying to control this relay.
My code as follows:

`{
  "c": {
    "io": [[0],2,0,0],
    "o": 0
    },
  "a": [
    {
    "t": 1,
    "0":{"r":[[0,0]]},
    "1":{"r":[[0,1]]},
    "s":0
    }
  ]
}`

The switch shows up in HomeKit and I can switch it there, but the GPIO-Pins (I also tried GPIO2) don't seem to react accordingly.
Thanks for the help!

**Comment by RokkakuM** [comment]
The old version of json looked like this but is no longer available on the new firmware, can you convert to the new json format?

{"c":{"l":1,"b":[{"g":2,"t":5}]},"a":[{"t":1,"s":1,"0":{"r":[{"g":0,"v":1}]},"1":{"r":[{"g":0,"v":0}]},"b":[{"g":2}]}]}

**Comment by Tellicious** [comment]
Actually the script looks fine apart from missing parentheses on `"io"` declaration, even though many options could be removed. You could try something like this:

`{"c":{"io":[[[0],2]]},"a":[{"0":{"r":[[0]]},"1":{"r":[[0,1]]},"s":0}]}`

**Reply by vifi42** [reply]
Thanks! That worked great!

**Reply by shengweitw** [reply]
{"c":{"o":0},"a":[{"t":3,"b":[],"s":0,"f0":[{"g":0,"t":1,"p":1}],"f1":[{"g":0,"t":2,"p":1}],"f2":[{"g":0,"t":3,"p":1}]},{"t":3,"b":[],"s":0,"f0":[{"g":2,"t":1,"p":1}],"f1":[{"g":2,"t":2,"p":1}],"f2":[{"g":2,"t":3,"p":1}]}]}

Excuse me, how to convert to the new JSON CODE.

**Reply by Tellicious** [reply]
I suggest you try converting it starting from my script above and from those 2 links that provide you also with good examples: 
- [Stateless Button](https://github.com/RavenSystem/esp-homekit-devices/wiki/stateless-button) 
- [GPIO Configuration](https://github.com/RavenSystem/esp-homekit-devices/wiki/gpios-configuration)

**Reply by shengweitw** [reply]
Thank you, I'll take a closer look. Also, let me ask you, do you not have an infrared kit? It is used to control the air conditioner.

**Reply by Tellicious** [reply]
Nope, unfortunately I do not have any relevant experience on that
